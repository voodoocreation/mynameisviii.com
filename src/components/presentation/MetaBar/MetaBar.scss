@import "../../../scss/variables", "../../../scss/mixins";

.MetaBar {
  @include panel-1();

  display: flex;
  flex-wrap: wrap;
  line-height: 1.5;
  margin-bottom: $spacing;
  padding: $spacing-half;

  > * {
    display: flex;
    align-items: center;
    position: relative;
    padding: 0 $spacing-half;

    > span {
      padding-left: ($spacing / 3);
    }
  }
}
