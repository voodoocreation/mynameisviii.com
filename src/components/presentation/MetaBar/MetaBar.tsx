import cn from "classnames";
import * as React from "react";

interface IProps {
  className?: string;
}

const MetaBar: React.SFC<IProps> = ({ children, className }) =>
  !children ? null : <div className={cn("MetaBar", className)}>{children}</div>;

export default MetaBar;
