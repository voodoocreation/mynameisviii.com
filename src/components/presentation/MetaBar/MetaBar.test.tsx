import WrapperWithIntl from "../../../utilities/WrapperWithIntl";
import MetaBar from "./MetaBar";

const component = new WrapperWithIntl(MetaBar).withDefaultProps({
  className: "TestMetaBar"
});

describe("[presentation] <MetaBar />", () => {
  it("doesn't render anything when there are no children", () => {
    const wrapper = component.render();

    expect(wrapper.html()).toBeNull();
  });

  describe("when children are defined", () => {
    const wrapper = component.withChildren("Child meta").render();

    it("renders the children within the container", () => {
      expect(wrapper.html()).toBe("Child meta");
    });

    it("matches snapshot", () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
