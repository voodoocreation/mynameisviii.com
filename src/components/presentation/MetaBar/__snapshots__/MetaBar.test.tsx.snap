// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[presentation] <MetaBar /> when children are defined matches snapshot 1`] = `
<div
  class="MetaBar TestMetaBar"
>
  Child meta
</div>
`;
